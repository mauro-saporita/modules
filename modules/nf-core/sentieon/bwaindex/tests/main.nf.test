nextflow_process {

    name "Test Process SENTIEON_BWAINDEX"
    script "../main.nf"
    process "SENTIEON_BWAINDEX"
    tag "modules"
    tag "modules_nfcore"
    tag "sentieon"
    tag "sentieon/bwaindex"

    test("test sentieon bwaindex") {

        when {
            process {
                """
                input[0] = [ [ id: 'test' ], file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true) ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
