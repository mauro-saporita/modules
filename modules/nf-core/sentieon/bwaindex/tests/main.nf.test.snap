{
    "versions": {
        "content": [
            [
                "versions.yml:md5,b889fa1ad19d9db7a5d00f005cedff4e"
            ]
        ],
        "timestamp": "2025-03-25T18:19:00.318515"
    },
    "test sentieon bwaindex": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genome.amb:md5,3a68b8b2287e07dd3f5f95f4344ba76e",
                            "genome.ann:md5,c32e11f6c859f166c7525a9c1d583567",
                            "genome.bwt:md5,0469c30a1e239dd08f68afe66fde99da",
                            "genome.pac:md5,983e3d2cd6f36e2546e6d25a0da78d66",
                            "genome.sa:md5,ab3952cabf026b48cd3eb5bccbb636d1"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,b889fa1ad19d9db7a5d00f005cedff4e"
                ],
                "index": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genome.amb:md5,3a68b8b2287e07dd3f5f95f4344ba76e",
                            "genome.ann:md5,c32e11f6c859f166c7525a9c1d583567",
                            "genome.bwt:md5,0469c30a1e239dd08f68afe66fde99da",
                            "genome.pac:md5,983e3d2cd6f36e2546e6d25a0da78d66",
                            "genome.sa:md5,ab3952cabf026b48cd3eb5bccbb636d1"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b889fa1ad19d9db7a5d00f005cedff4e"
                ]
            }
        ],
        "timestamp": "2025-03-25T18:19:00.298436"
    }
}